angular/src/components/change-password.component.ts

```diff
@@ -40,7 +40,7 @@ export class ChangePasswordComponent implements OnInit {
 
   async ngOnInit() {
     this.email = await this.stateService.getEmail();
-    this.enforcedPolicyOptions = await this.policyService.getMasterPasswordPolicyOptions();
+    this.enforcedPolicyOptions ??= await this.policyService.getMasterPasswordPolicyOptions();
```

Comment:
If I'm understanding this correctly, this assigns the result of `this.policyService.getMasterPasswordPolicyOptions()` to `enforcedPolicyOptions` if it is falsy, right? Wouldn't it always be falsy on `ngOnInit` if it is not passed as an input?

Reply:
Correct, inside of the `ngOnInit` for `set-password.component.ts` we detect if we're registering a new user into an organization and if so fill `enforcedPolicyOptions`:
```ts
// Automatic Enrollment Detection
    if (this.identifier != null) {
      try {
        const response = await this.apiService.getOrganizationAutoEnrollStatus(this.identifier);
        this.orgId = response.id;
        this.resetPasswordAutoEnroll = response.resetPasswordEnabled;
        this.enforcedPolicyOptions =
          await this.policyService.getMasterPasswordPoliciesForInvitedUsers(this.orgId);
      } catch {
        this.platformUtilsService.showToast("error", null, this.i18nService.t("errorOccurred"));
      }
    }
```

The reason I check if its falsy is to make sure it hasn't been previously loaded. Unless I'm misunderstanding the order in which those are run?

Reply:
Ah that makes sense! I just looked and it looks like `super.ngOnInit()` is in fact called the logic in `set-password.component.ts` 👍